Dockerfile-rust

```diff
@@ -2,21 +2,21 @@ FROM rust:latest
 LABEL maintainer="Fastly OSS <oss@fastly.com>"
 
 ENV RUST_TOOLCHAIN=stable
-RUN rustup toolchain install ${RUST_TOOLCHAIN}
-RUN rustup target add wasm32-wasi --toolchain ${RUST_TOOLCHAIN}
-
-RUN apt-get update && apt-get install -y curl jq
-RUN export FASTLY_CLI_VERSION=$(curl --silent https://api.github.com/repos/fastly/cli/releases/latest | jq -r .tag_name | cut -d 'v' -f 2) \
-  GOARCH=$(dpkg --print-architecture) \
-  && curl -vL "https://github.com/fastly/cli/releases/download/v${FASTLY_CLI_VERSION}/fastly_v${FASTLY_CLI_VERSION}_linux-$GOARCH.tar.gz" -o fastly.tar.gz \
-  && curl -vL "https://github.com/fastly/cli/releases/download/v${FASTLY_CLI_VERSION}/fastly_v${FASTLY_CLI_VERSION}_SHA256SUMS" -o sha256sums \
+RUN rustup toolchain install ${RUST_TOOLCHAIN} \
+  && rustup target add wasm32-wasi --toolchain ${RUST_TOOLCHAIN} \
+  && apt-get update && apt-get install -y curl jq && apt-get -y clean && rm -rf /var/lib/apt/lists/* \
+  && export FASTLY_CLI_VERSION=$(curl -s https://api.github.com/repos/fastly/cli/releases/latest | jq -r .tag_name | cut -d 'v' -f 2) \
+            GOARCH=$(dpkg --print-architecture) \
+  && curl -sL "https://github.com/fastly/cli/releases/download/v${FASTLY_CLI_VERSION}/fastly_v${FASTLY_CLI_VERSION}_linux-$GOARCH.tar.gz" -o fastly.tar.gz \
+  && curl -sL "https://github.com/fastly/cli/releases/download/v${FASTLY_CLI_VERSION}/fastly_v${FASTLY_CLI_VERSION}_SHA256SUMS" -o sha256sums \
   && dlsha=$(shasum -a 256 fastly.tar.gz | cut -d " " -f 1) && expected=$(cat sha256sums | awk -v pat="$dlsha" '$0~pat' | cut -d " " -f 1) \
-  && if [[ "$dlsha" != "$expected" ]]; then echo "shasums don't match" && exit 1; fi
-
-RUN tar -xzf fastly.tar.gz --directory /usr/bin && rm fastly.tar.gz
+  && if [ "$dlsha" != "$expected" ]; then echo "shasums don't match" && exit 1; fi \
+  && tar -xzf fastly.tar.gz --directory /usr/bin && rm -f sha256sums fastly.tar.gz \
+  && useradd -ms /bin/bash fastly
 
-RUN adduser fastly
 USER fastly
+# this forces the download of the crates index; it will save build time if the image was recently built
+RUN cargo search --limit 0
```

Comment:
Outside the concern of this PR but just thinking we should really have a `Dockerfile-go` as well.

Reply:
+1!! Our cli depends on the specific-programming-language build tooling, and I dont think that a common image bringing all languages tooling is going to scale well (will make the image even more huge). What we may want to do is to consolidate all Dockerfiles in a single one: a common stage can just install the fastly-cli, and then we can have dedicated stages per language we are supporting, each stage picking the common fastly-cli and adding the language-specific tooling (or inheriting it from the right base image). Then, rather than doing a docker-build against a specific Dockerfile, CI and users will need to be explicit on the stage we want to build (the common fastly-cli layers will be reused if you build more than one image/language, which is good for caching)

Reply:
@dcaba Nice. I bow to your knowledge in this area 🙇🏻  (ps, PRs welcome 😛)